lib/execjs/support/node_runner.js

```diff
@@ -1,10 +1,20 @@
-(function(program, execJS) { execJS(program) })(function(global, module, exports, require, console, setTimeout, setInterval, clearTimeout, clearInterval, setImmediate, clearImmediate) { #{source}
+(function(program, execJS) { execJS(program) })(function(global, process, module, exports, require, console, setTimeout, setInterval, clearTimeout, clearInterval, setImmediate, clearImmediate) { #{source}
 }, function(program) {
   var output, print = function(string) {
     process.stdout.write('' + string);
   };
   try {
+    var __process__ = process;
+    delete this.process;
+    delete this.console;
+    delete this.setTimeout;
+    delete this.setInterval;
+    delete this.clearTimeout;
+    delete this.clearInterval;
+    delete this.setImmediate;
+    delete this.clearImmediate;
     result = program();
```

Comment:
Someday we could move Nodes execution over to `vm.Script`. We'd have full control over what globals were exposed to the v8 context. `vm.Script` also has nice support for timeouts.